portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,41 +31,42 @@
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
-public class DataServlet extends HttpServlet {
-  private final List<String> comments = new ArrayList<>();
 
+public class DataServlet extends HttpServlet {
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     String currComment = request.getParameter("comment");
     writeComments(currComment);
-    response.sendRedirect("/index.html");
-
-    // String currcomment = getParameter(request, "comment", "");
-    // comments.add(currcomment);
-    // response.setContentType("application/json;");
-    // String json = convertToJson(comments);
-    // response.getWriter().println(json);
+    List<String> comments = readComments();
+    response.setContentType("application/json;");
```

Comment:
Since you don't need the comments variable for anything other than passing it to the convertToJson method, you can remove this variable and directly do convertToJson(readComments();

Reply:
Done